queue_services/activity-log-listener/src/activity_log_listener/config.py

```diff
@@ -0,0 +1,138 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""All of the configuration for the service is captured here.
+
+All items are loaded, or have Constants defined here that
+are loaded into the Flask configuration.
+All modules and lookups get their configuration from the
+Flask config, rather than reading environment variables directly
+or by accessing this configuration directly.
+"""
+import os
+import random
+
+from dotenv import find_dotenv, load_dotenv
+
+
+# this will load all the envars from a .env file located in the project root (api)
+load_dotenv(find_dotenv())
+
+CONFIGURATION = {
+    'development': 'activity_log_listener.config.DevConfig',
+    'testing': 'activity_log_listener.config.TestConfig',
+    'production': 'activity_log_listener.config.ProdConfig',
+    'default': 'activity_log_listener.config.ProdConfig'
+}
+
+
+def get_named_config(config_name: str = 'production'):
+    """Return the configuration object based on the name.
+
+    :raise: KeyError: if an unknown configuration is requested
+    """
+    if config_name in ['production', 'staging', 'default']:
+        app_config = ProdConfig()
+    elif config_name == 'testing':
+        app_config = TestConfig()
+    elif config_name == 'development':
+        app_config = DevConfig()
+    else:
+        raise KeyError(f'Unknown configuration: {config_name}')
+    return app_config
+
+
+class _Config():  # pylint: disable=too-few-public-methods
+    """Base class configuration that should set reasonable defaults.
+
+    Used as the base for all the other configurations.
+    """
+
+    PROJECT_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+    SENTRY_DSN = os.getenv('SENTRY_DSN', None)
+
+    SQLALCHEMY_TRACK_MODIFICATIONS = False
+
+    # POSTGRESQL
+    DB_USER = os.getenv('DATABASE_USERNAME', '')
+    DB_PASSWORD = os.getenv('DATABASE_PASSWORD', '')
+    DB_NAME = os.getenv('DATABASE_NAME', '')
+    DB_HOST = os.getenv('DATABASE_HOST', '')
+    DB_PORT = os.getenv('DATABASE_PORT', '5432')
+    SQLALCHEMY_DATABASE_URI = 'postgresql://{user}:{password}@{host}:{port}/{name}'.format(
+        user=DB_USER,
+        password=DB_PASSWORD,
+        host=DB_HOST,
+        port=int(DB_PORT),
+        name=DB_NAME,
+    )
+
+    NATS_CONNECTION_OPTIONS = {
+        'servers': os.getenv('NATS_SERVERS', 'nats://127.0.0.1:4222').split(','),
+        'name': os.getenv('NATS_ACCOUNT_CLIENT_NAME', 'accounts.activity_log.worker')
+
+    }
+    STAN_CONNECTION_OPTIONS = {
+        'cluster_id': os.getenv('NATS_CLUSTER_ID', 'test-cluster'),
+        'client_id': str(random.SystemRandom().getrandbits(0x58)),
+        'ping_interval': 1,
+        'ping_max_out': 5,
+    }
+
+    SUBSCRIPTION_OPTIONS = {
+        'subject': os.getenv('NATS_ACCOUNT_SUBJECT', 'account.activity.events'),
+        'queue': os.getenv('NATS_ACCOUNT_QUEUE', 'ccount.events.worker'),
```

Comment:
'a' is missing from default queue name. Also shouldn't this use a different durable name? As it may cause the account events listener to not receive any message.

Reply:
changed